managed/src/main/java/com/yugabyte/yw/controllers/EncryptionAtRestController.java

```diff
@@ -175,33 +172,31 @@ public Result retrieveKey(UUID customerUUID, UUID universeUUID) {
                 customerUUID.toString(),
                 universeUUID.toString()
         ));
-        byte[] keyRef = null;
-        byte[] recoveredKey = null;
-        try {
-            ObjectNode formData = (ObjectNode) request().body().asJson();
-            keyRef = Base64.getDecoder().decode(formData.get("reference").asText());
-            UUID configUUID = UUID.fromString(formData.get("configUUID").asText());
-            recoveredKey = keyManager.getUniverseKey(universeUUID, configUUID, keyRef);
-            if (recoveredKey == null || recoveredKey.length == 0) {
-                final String errMsg = String.format(
-                        "No universe key found for universe %s",
-                        universeUUID.toString()
-                );
-                throw new RuntimeException(errMsg);
-            }
-            ObjectNode result = Json.newObject()
-                    .put("reference", keyRef)
-                    .put("value", Base64.getEncoder().encodeToString(recoveredKey));
-            auditService().createAuditEntry(ctx(), request(), formData);
-            return ApiResponse.success(result);
-        } catch (Exception e) {
-            final String errMsg = String.format(
-                    "Could not recover universe key from universe %s",
-                    universeUUID.toString()
-            );
-            LOG.error(errMsg, e);
-            return ApiResponse.error(BAD_REQUEST, e.getMessage());
-        }
+        ObjectNode formData = (ObjectNode) request().body().asJson();
+        byte[] keyRef = Base64.getDecoder().decode(formData.get("reference").asText());
+        UUID configUUID = UUID.fromString(formData.get("configUUID").asText());
+        byte[] recoveredKey = getRecoveredKeyOrBadRequest(
+          universeUUID,
+          configUUID,
+          keyRef
+        );
+        ObjectNode result = Json.newObject()
+          .put("reference", keyRef)
+          .put("value", Base64.getEncoder().encodeToString(recoveredKey));
+        auditService().createAuditEntry(ctx(), request(), formData);
+        return ApiResponse.success(result);
+    }
+
+    public byte[] getRecoveredKeyOrBadRequest(UUID universeUUID, UUID configUUID, byte[] keyRef) {
+      byte[] recoveredKey = keyManager.getUniverseKey(universeUUID, configUUID, keyRef);
```

Comment:
I meant make KeyManager.getRecoveredKeyOrBadRequest instead of  KeyManager.getUniverseKey

Reply:
I already tried that but its not working, it seems to be tricky. Will take care of this in the followup PR. Thanks!
Merging this for now.